q2cli/builtin/dev.py

```diff
@@ -141,57 +143,93 @@ def reset_theme():
-              type=click.Path(exists=True, file_okay=True,
-                              dir_okay=False, readable=True),
-              help='Lorem Ipsum')
-@click.option('--target-path', required=True,
-              help='The path to the target file')
+@click.argument('input-path', type=click.Path(exists=True, file_okay=True,
+                dir_okay=False, readable=True),
+                metavar=_COMBO_METAVAR)
+@click.option('--zip-data-path', required=True,
+              help='The path within the zipped Result\'s data/'
+                   ' directory that should be searched.')
 @click.option('--expression', required=True,
-              help='The line to match')
-def assert_has_line(input_path, target_path, expression):
+              help='The Python regular expression to match.')
+def assert_result_data(input_path, zip_data_path, expression):
     import re
     import q2cli.util
     import qiime2.sdk
+    from q2cli.core.config import CONFIG
 
     q2cli.util.get_plugin_manager()
-    result = qiime2.sdk.Result.load(input_path)
-
-    hits = sorted(result._archiver.data_dir.glob(target_path))
-    if len(hits) != 1:
-        raise ValueError('Value provided for target_path (%s) did not produce '
-                         'exactly one hit: %s' % (target_path, hits))
-
-    target = hits[0].read_text()
-    match = re.search(expression, target, flags=re.MULTILINE)
-    if match is None:
-        raise AssertionError('Expression %r not found in %s.' %
-                             (expression, target_path))
+
+    try:
+        result = qiime2.sdk.Result.load(input_path)
+    except Exception as e:
+        header = 'There was a problem loading %s as a QIIME 2 result:' % \
+                input_path
+        q2cli.util.exit_with_error(e, header=header)
+
+    try:
+        hits = sorted(result._archiver.data_dir.glob(zip_data_path))
+        if len(hits) != 1:
+            data_dir = result._archiver.data_dir
+            all_fps = sorted(data_dir.glob('**/*'))
+            all_fps = [x.relative_to(data_dir).name for x in all_fps]
+            raise ValueError('Value provided for zip_data_path (%s) did not '
+                             'produce exactly one match.\nMatches: %s\n'
+                             'Paths observed: %s' %
+                             (zip_data_path, hits, all_fps))
+    except Exception as e:
+        header = 'There was a problem locating the zip_data_path (%s)' % \
+                zip_data_path
+        q2cli.util.exit_with_error(e, header=header)
+
+    try:
+        target = hits[0].read_text()
+        match = re.search(expression, target, flags=re.MULTILINE)
+        if match is None:
+            raise AssertionError('Expression %r not found in %s.' %
+                                 (expression, zip_data_path))
+    except Exception as e:
+        header = 'An unexpected error occurred while running'
+        ' "assert-result-data"'
```

Comment:
![image](https://user-images.githubusercontent.com/274668/140760206-57c794a4-bb03-418b-82c0-d8aabaade682.png)

This isn't an "error" per-se, let's match what you wrote above:

```suggestion
        header = 'There was a problem finding the expression:'
```